engine/lib/comments.php

```diff
@@ -193,3 +194,36 @@ function _elgg_comments_permissions_override($hook, $type, $return, $params) {
 	
 	return $return;
 }
+
+/**
+ * Set subject for email notifications about new ElggComment objects
+ *
+ * The "Re: " part is required by some email clients in order to properly
+ * group the notifications in threads.
+ *
+ * Group discussion replies extend ElggComment objects so this takes care
+ * of their notifications also.
+ *
+ * @param string $hook        'email'
+ * @param string $type        'system'
+ * @param array  $returnvalue Current mail parameters
+ * @param array  $params      Original mail parameters
+ * @return array $returnvalue Modified mail parameters
+ */
+function _elgg_comments_notification_email_subject($hook, $type, $returnvalue, $params) {
+
+	/** @var Elgg_Notifications_Notification */
+	$notification = elgg_extract('notification', $returnvalue['params']);
+
+	if ($notification instanceof Elgg_Notifications_Notification) {
```

Comment:
fix #7783. We should definitely watch out for anything like "Elgg_[A-Z]" while merging into 1.10